perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2216,6 +2216,9 @@ public Member setSponsorshipForMember(PerunSession session, Member sponsoredMemb
 		getPerunBl().getAuditer().log(session, new SponsoredMemberSet(sponsoredMember));
 		getPerunBl().getAuditer().log(session, new SponsorshipEstablished(sponsoredMember, sponsor, validityTo));
 
+		//validate sponsored member in the VO (you can do it asynchronous) - if member was already valid, it will not change anything
+		validateMemberAsync(session, sponsoredMember);
```

Comment:
This is about existing member, right? What about membership expiration? Above methods do nothing with it, so the member would be switched back to expired after midnight if his expiration date is in the past.

Also I'm not sure if we don't want to have also "sync" validation for the purpose of the tests, since it creates thread outside of the test scope (rollback), but tests passed, since they test only sponsorship validity, not members validity. @martin-kuba ?

Reply:
Right I will check the expiration too.
There is no problem to do it synchronously instead, I just wanted to save the time of sponsor here.

Reply:
I changed that to copy logic from sponsorMember method.